shared/wallets/send-form/asset-input/asset-input-advanced.tsx

```diff
@@ -101,31 +102,111 @@ export const AssetInputSenderAdvanced = (props: EmptyProps) => (
 )
 
 export const AssetPathIntermediate = () => {
+  const [expanded, setExpanded] = React.useState(false)
   const path = Container.useSelector(state => state.wallets.builtPaymentAdvanced.fullPath.path)
-  return !path.size ? (
-    <Kb.Divider />
-  ) : (
-    <Kb.Box2 direction="horizontal" style={styles.assetPathContainer} fullWidth={true}>
-      <Kb.Text type="BodySmallSemibold">Intermediate asset(s): </Kb.Text>
-      <Kb.Box style={Styles.globalStyles.flexGrow} />
-      <Kb.Box2 direction="vertical" centerChildren={true} style={styles.assetPathItem} gap="xtiny">
-        {path
-          .toArray()
-          .reverse()
-          .map(asset => (
-            <React.Fragment key={Types.assetDescriptionToAssetID(asset)}>
-              <Kb.Icon type="iconfont-arrow-full-up" sizeType="Tiny" />
-              <Kb.Box2 direction="horizontal">
-                <Kb.Text type="BodyTinyBold">{asset === 'native' ? 'XLM' : `${asset.code}`}</Kb.Text>
-                <Kb.Text type="BodyTiny" lineClamp={1} ellipsizeMode="middle">
-                  {asset !== 'native' && `/${asset.issuerVerifiedDomain || asset.issuerAccountID}`}
-                </Kb.Text>
-              </Kb.Box2>
-            </React.Fragment>
-          ))}
-        <Kb.Icon type="iconfont-arrow-full-up" sizeType="Tiny" />
+  if (!path.size) {
+    return <Kb.Divider />
+  }
+  return (
+    <Kb.Box
+      style={Styles.collapseStyles([
+        styles.intermediateContainer,
+        expanded && styles.intermediateContainerExpanded,
+        !expanded && styles.intermediateContainerCollapsed,
+      ])}
+    >
+      <Kb.Box2
+        direction="vertical"
+        alignItems="center"
+        style={Styles.collapseStyles([
+          styles.intermediateAbsoluteBlock,
```

Comment:
seems like this could all be inside a `styles` attribute.. is there a reason to prefer doing the arithmetic inline?

Reply:
not really; just that it was so small but I guess that's not a good reason. Will move into `styles`